src/util/util.ts

```diff
@@ -1,75 +1,186 @@
+    return {
+      spec: spec,
+      label: name,
+      detail: desc,
+    };
+  });
+
+  const specResults: MPSpecOption[] | undefined = await vscode.window.showQuickPick(mpSpecOptions, {
+    ignoreFocusOut: true,
+    canPickMany: true,
+    placeHolder: "Select MicroProfile specifications.",
+  });
+
+  if (specResults != null) {
+    return specResults.map(result => result.spec);
+  }
+  return undefined;
 }
 
-export async function mapToDescription(prevArray : string[], descriptions: Object) {
-	var specItems : vscode.QuickPickItem[] = new Array();
-	prevArray.forEach(element => {
-		var content = descriptions[element as keyof typeof descriptions];
-		var contentStr = new String(content);
-
-		var splits = contentStr.split(" - ");
-		specItems.push({
-			'label': splits[0],
-			// 'detail': capitalizeFirstLetter(splits[1])
-			'detail': splits[1]
-		});
-	});
-	return specItems;
+export async function askForFolder(customOptions: OpenDialogOptions): Promise<Uri | undefined> {
+  const options: OpenDialogOptions = {
+    canSelectFiles: false,
+    canSelectFolders: true,
+    canSelectMany: false,
+  };
+  const result = await window.showOpenDialog(Object.assign(options, customOptions));
+
+  if (result && result.length > 0) {
+    return result[0];
+  }
+
+  return undefined;
 }
 
-export function getKeyFromValue(value: Object, arr : any) {
-	return Object.keys(arr).find(key => arr[key] === value);
+interface RequestOptions {
+  url: string;
 }
 
+// Downloads a file using streams to avoid loading entire file into memowry
+export async function downloadFile(
+  requestOptions: RequestOptions,
+  downloadLocation: string
+): Promise<void> {
+  const { url, ...options } = requestOptions;
+  const res = await fetch(url, options);
+  if (res.status >= 400 && res.status < 600) {
+    throw new Error(`Bad response from server ${res.status}: ${res.statusText}`);
+  }
+
+  return new Promise((resolve, reject) => {
+    // create a pipeline that pipes the response to the download location
+    pipeline(res.body, fs.createWriteStream(downloadLocation), err => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
 
-export async function resolveSpecs(specifications : String[], descriptions: Object){
-	var dataString: String = "";
-	if (specifications !== undefined && specifications.length > 0) {
-		dataString += ', "selectedSpecs":[';
-		specifications.forEach(async element => {
-			var apiSpec;
-			if (element === specifications[0]){
-				apiSpec = getKeyFromValue(element, descriptions);
-				dataString = dataString + '"' + apiSpec + '"';
-			} else {
-				apiSpec = getKeyFromValue(element, descriptions);
-				dataString += ', "' + apiSpec + '"';
-			}
-		});
-		dataString = dataString + ']';
-	}
-	return dataString;
+export function capitalizeFirstLetter(newString: string): string {
+  return newString.charAt(0).toUpperCase() + newString.slice(1);
 }
```

Comment:
trim the string with `newString = newString.trim()` to remove leading whitespace